#! /bin/sh /usr/share/dpatch/dpatch-run
## add-libdst.a-dhcp3-dev.dpatch by  <apollock@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patch from Olivier Houchard for #220525
## DP: Makes the install target for libdst.a more useful than the default

@DPATCH@
diff -urNad dhcp3-3.0.2/dst/Makefile.dist /tmp/dpep.Zd4nUc/dhcp3-3.0.2/dst/Makefile.dist
--- dhcp3-3.0.2/dst/Makefile.dist	2004-06-11 03:59:28.000000000 +1000
+++ /tmp/dpep.Zd4nUc/dhcp3-3.0.2/dst/Makefile.dist	2005-06-15 20:35:54.133359161 +1000
@@ -30,13 +30,25 @@
 
 all:	libdst.a
 
-install:
-
 libdst.a:	$(OBJ)
 	rm -f dst.a
 	ar cruv libdst.a $(OBJ)
 	$(RANLIB) libdst.a
 
+install: all
+	for dir in $(LIBDIR);\
+	 do \
+	  foo=""; \
+	  for bar in `echo $(DESTDIR)$${dir} |tr / ' '`; do \
+	    foo=$${foo}/$$bar; \
+	    if [ ! -d $$foo ]; then \
+	      mkdir $$foo; \
+	      chmod 755 $$foo; \
+	    fi; \
+	  done; \
+	done
+	$(INSTALL) libdst.a $(DESTDIR)$(LIBDIR)
+
 depend:
 	$(MKDEP) $(INCLUDES) $(PREDEFINES) $(SRC)
 
